bokehjs/src/lib/core/util/pretty.ts

```diff
@@ -18,12 +18,20 @@ export type PrinterOptions = {
 
 export class Printer {
   readonly precision?: number
+  protected readonly visited: Set<object> = new Set()
 
   constructor(options?: PrinterOptions) {
     this.precision = options?.precision
   }
 
   to_string(obj: unknown): string {
+    if (isObject(obj)) {
+      if (this.visited.has(obj))
+        return "<circular>"
```

Comment:
slight preference to `<cycle>`  here (noun instead of adjective)

Reply:
I'm using node's nomenclature here:
```
$ node
Welcome to Node.js v14.14.0.
Type ".help" for more information.
> x = {y: 1}
{ y: 1 }
> x.z = x
<ref *1> { y: 1, z: [Circular *1] }
```